tests/compile-fail/transmute.rs

```diff
@@ -0,0 +1,38 @@
+#![feature(core)]
+#![feature(plugin)]
+#![plugin(clippy)]
+
+extern crate core;
+
+use std::mem::transmute as my_transmute;
+use std::vec::Vec as MyVec;
+
+fn my_vec() -> MyVec<i32> {
+    vec![]
+}
+
```

Comment:
It's not possible to generically transmute from `T` to `U`:
```
error: cannot transmute to or from a type that contains unsubstituted type parameters [E0139]
```
but I'll add a test from `&T` to `&U`.